dart/lib/src/transport/rate_limit_parser.dart

```diff
@@ -0,0 +1,63 @@
+import 'rate_limit_category.dart';
+import 'rate_limit.dart';
+
+/// Parse rate limit categories and times from response header payloads.
+class RateLimitParser {
+  RateLimitParser(this._header);
+
+  static const httpRetryAfterDefaultDelay = Duration(milliseconds: 60000);
+
+  final String? _header;
+
+  List<RateLimit> parseRateLimitHeader() {
+    final rateLimitHeader = _header;
+    if (rateLimitHeader == null) {
+      return [];
+    }
+
+    final rateLimits = <RateLimit>[];
+
+    final rateLimitValues = rateLimitHeader.toLowerCase().split(',');
+    rateLimitValues.forEach((rateLimitValue) {
```

Comment:
I feel like the indentation here can be reduced by using a for loop and `continue`s.